Clean puppet reports

davidoae opened the pull request:
We're keeping puppet run logs for ever and they use space and we don't need them so this sets a cronjob to remove puppet reports older than 30 days.

simong commented on modules/oaeservice/manifests/puppet.pp:
```diff
@@ -7,4 +7,15 @@
+  # the report dir is not in puppet-hilary anywhere so extracted direct from puppet.conf
+  cron { 'clean-puppet-reports':
+    ensure  => present,
+    command => "find $(awk -F= '/^reportdir/ { print $2 }' /etc/puppet/puppet.conf) -type f -iname \*.yaml -ctime +30 -delete",
```
Can this directory be defaulted to `/var/lib/puppet/reports` somehow? The QA pupper server doesn't seem to configure a reportdir

davidoae replied:
I started there, but in production the reports are in /data/puppet/reports, hence the awk. I didn't check qa.
Honestly this is such a minor thing that I should just add the reportdir config into qa's puppet.conf.
I only noticed this issue 'cos puppet in staging is almost out of space, mostly these reports :)

davidoae replied:
Would need to add reportdir to puppmaster provision script too.

simong commented:
Back to @davidoae for follow-up

mrvisser commented:
Just a caution that we do in fact have a cronjob on the production puppet master that does this:

`0 1 * * * find /data/puppet/reports -name "*.yaml" -mtime +10 -exec rm -f {} \;`

However it's not managed by puppet, but it should. So be sure to remove the unmanaged one when deploying.

Thanks for moving this forward!

davidoae commented:
Thanks dude, I completely missed that one heh.


On 4 June 2015 at 12:49, Branden Visser <notifications@github.com> wrote:

> Just a caution that we do in fact have a cronjob on the production puppet
> master that does this:
>
> 0 1 * * * find /data/puppet/reports -name "*.yaml" -mtime +10 -exec rm -f
> {} \;
>
> However it's not managed by puppet, but it should. So be sure to remove
> the unmanaged one when deploying.
>
> Thanks for moving this forward!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/oaeproject/puppet-hilary/pull/158#issuecomment-108857162>
> .
>

davidoae commented:
I have added the change to provision script.
The cron that's currently in place I'll remove manually after merge.
I've added reportdir to qa's puppet.conf so merge shouldn't break qa.
